pyproject.toml

```diff
@@ -0,0 +1,38 @@
+[project]
+name = "pems"
+version = "0.0.1"
+description = "Caltrans Performance Measurement System (PeMS) is an application that enables access to traffic data collected by sensors that span the freeway system across all major metropolitan areas of the State of California."
+readme = "README.md"
+license = { file = "LICENSE" }
+classifiers = ["Programming Language :: Python :: 3 :: Only"]
+requires-python = ">=3.12"
+maintainers = [
+  { name = "Compiler LLC", email = "dev@compiler.la" }
+]
+dependencies = [
+]
+
+[project.optional-dependencies]
+dev = [
+    "black",
+    "flake8",
+    "pre-commit"
+]
+
+[project.urls]
+Code = "https://github.com/compilerla/pems"
+Homepage = "https://pems.dot.ca.gov/"
+Issues = "https://github.com/compilerla/pems/issues"
+
+[build-system]
+requires = ["setuptools>=75"]
+build-backend = "setuptools.build_meta"
+
+[tool.black]
+line-length = 127
+target-version = ['py311']
```

Comment:
I think in the past there wasn't a `py312` available, but maybe there is now? That would make more sense since we are working on Python 3.12